### PR TITLE
Don't allow empty env names

### PR DESCRIPTION
**Pull request by duglin**
Closes: #25281

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: builder/dockerfile/parser/line_parsers.go [hunk 1]
```diff
@@ -168,6 +168,10 @@ func parseNameVal(rest string, key string, d *Directive) (*Node, map[string]bool
 			}
 			parts := strings.SplitN(word, "=", 2)
 
+			if len(parts[0]) == 0 {
+				return nil, nil, fmt.Errorf("Syntax error - ENV names can't be blank")
```
Comment (vdemeester) [inline builder/dockerfile/parser/line_parsers.go]: `ENV` shouldn't be hardcoded here though (it's used for other directive). Probably should be `fmt.Errorf("Syntax error - %s names can't be blank", key)`

Comment (duglin) [inline builder/dockerfile/parser/line_parsers.go]: yea, never mind - I'm redoing this - I have a better spot to do the check

**Comment by duglin** [comment]
ok- updated with a better set of checks

**Comment by vdemeester** [comment]
@duglin yes ! :+1: 
LGTM 🐸

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**